cpp/benchmarks/reshape/stack_benchmark.cu

```diff
@@ -0,0 +1,204 @@
+
+
+
+
+class String : public ::benchmark::Fixture {};
+
+void BM_mar(benchmark::State& state){
+  using wrapper = cudf::test::column_wrapper<cudf::nvstring_category>;
+
+  const gdf_size_type num_columns{(gdf_size_type)state.range(0)};
+  const gdf_size_type column_size{(gdf_size_type)state.range(1)};
+
+  std::vector<wrapper> key_columns;
+
+  cudf::test::column_wrapper_factory<cudf::nvstring_category> factory;
+
+  for (gdf_size_type i = 0; i < num_columns; i++) {
+    key_columns.emplace_back(factory.make(column_size,
+      [=](gdf_index_type row) { return random_int(0, 10); }
+      // ,[=](gdf_index_type row) { return random_int(0, 10) < 90; }
+    ));
+  }
+
+  std::vector<gdf_column*> raw_cols(num_columns, nullptr);
+  std::transform(key_columns.begin(), key_columns.end(), raw_cols.begin(),
+                  [](wrapper &c) { return c.get(); });
+
+  for(auto _ : state){
+    cuda_event_timer timer(state, true);
+    NVCategory * combined_category = static_cast<NVCategory *>(raw_cols[0]->dtype_info.category);
+
+    for(int column_index = 1; column_index < num_columns; column_index++){
+      NVCategory * temp = combined_category;
+      if(raw_cols[column_index]->size > 0){
+        gdf_column* in_col = const_cast<gdf_column*>(raw_cols[column_index]);
+        combined_category = combined_category->merge_and_remap(
+            * static_cast<NVCategory *>(
+                in_col->dtype_info.category));
+        if(column_index > 1){
+          NVCategory::destroy(temp);
+        }
+      }
+    }
+
+    NVCategory::destroy(combined_category);
+  }
+}
+
+BENCHMARK_DEFINE_F(String, MAR)(::benchmark::State& state) {
+  BM_mar(state);
+}
+
+// BENCHMARK_REGISTER_F(String, MAR)
+//   ->UseManualTime()
+//   ->Unit(benchmark::kMillisecond)
+//   ->Apply(CustomArguments);
+
+
+void BM_cfc(benchmark::State& state){
+  using wrapper = cudf::test::column_wrapper<cudf::nvstring_category>;
+
+  const gdf_size_type num_columns{(gdf_size_type)state.range(0)};
+  const gdf_size_type column_size{(gdf_size_type)state.range(1)};
+
+  std::vector<wrapper> key_columns;
+
+  cudf::test::column_wrapper_factory<cudf::nvstring_category> factory;
+
+  for (gdf_size_type i = 0; i < num_columns; i++) {
+    key_columns.emplace_back(factory.make(column_size,
+      [=](gdf_index_type row) { return random_int(0, 10); }
+      // ,[=](gdf_index_type row) { return random_int(0, 10) < 90; }
+    ));
+  }
+
+  std::vector<gdf_column*> raw_cols(num_columns, nullptr);
+  std::transform(key_columns.begin(), key_columns.end(), raw_cols.begin(),
+                  [](wrapper &c) { return c.get(); });
+
+  for(auto _ : state){
+    cuda_event_timer timer(state, true);
+    std::vector<NVCategory *> cats;
+    std::transform(raw_cols.begin(), raw_cols.end(), std::back_inserter(cats), 
+      [&](gdf_column* c) { return static_cast<NVCategory *>(c->dtype_info.category); });
+
+    NVCategory * combined_category = NVCategory::create_from_categories(cats);
```

Comment:
I think NVCategory is going away...

Reply:
Yes, it is. That's why I said this PR would've been DOA so I was putting it on hold. But we decided to go through with it anyway because @randerzander asked for it now.